src/toil_lib/tools/variant_manipulation.py

```diff
@@ -0,0 +1,335 @@
+    :param JobFunctionWrappingJob job: passed automatically by Toil
+    :param str mode: variant type (SNP or INDEL)
+    :param str vcf_id: VCF FileStoreID
+    :param Namespace config: Configuration parameters and shared FileStoreIDs
+    :return: Filtered VCF FileStoreID
+    :rtype: str
+    """
+    mode = mode.upper()
+    job.fileStore.logToMaster('Applied %s Hard Filter' % mode)
+
+    inputs = {'genome.fa': config.genome_fasta,
+              'genome.fa.fai': config.genome_fai,
+              'genome.dict': config.genome_dict,
+              'input.vcf': vcf_id}
+
+    work_dir = job.fileStore.getLocalTempDir()
+    for name, file_store_id in inputs.iteritems():
+        job.fileStore.readGlobalFile(file_store_id, os.path.join(work_dir, name))
+
+    # GATK hard filters:
+    # https://software.broadinstitute.org/gatk/documentation/article?id=2806
+    # Filters are different for SNPs and INDELs. Please refer to GATK documentation.
+    if mode == 'SNP':
+        expression = '"QD < 2.0 || FS > 60.0 || MQ < 40.0 || ' \
+                     'MQRankSum < -12.5 || ReadPosRankSum < -8.0"'
+
+    elif mode == 'INDEL':
+        expression = '"QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0"'
+
+    else:
+        raise ValueError('Variant filter modes can be SNP or INDEL, got %s' % mode)
+
+    command = ['-T', 'VariantFiltration',
+               '-R', 'genome.fa',
+               '-V', 'input.vcf',
+               '--filterExpression', expression,
+               '--filterName', 'GATK_Germline_Hard_Filter_%s' % mode,   # Documents filter in header
+               '-o', 'filtered_variants.vcf']
+
+    docker_call(work_dir=work_dir,
+                env={'JAVA_OPTS': '-Djava.io.tmpdir=/data/ -Xmx{}'.format(job.memory)},
+                parameters=command,
+                tool='quay.io/ucsc_cgl/gatk:3.5--dba6dae49156168a909c43330350c6161dc7ecc2',
+                inputs=inputs.keys(),
+                outputs={'filtered_variants.vcf': None})
+
+    # Fix extra quotation marks in FILTER line
+    output = os.path.join(work_dir, 'filtered_variants.vcf')
+    sed_cmd = 's/"{}"/{}/'.format(expression, expression)
+    subprocess.call(['sed', '-i', sed_cmd, output])
+
+    return job.fileStore.writeGlobalFile(output)
+
+
+def gatk_variant_recalibrator(job, mode, vcf_id, config):
+    """
+    Variant quality score recalibration for SNP or INDEL variants. This function is configured
+    to run either SNP or INDEL VQSR, so run this function twice for the two variant types. It is
+    not necessary to split the input VCF by variant type to run this function.
+
+    :param JobFunctionWrappingJob job: Job instance
+    :param str mode: Determines variant recalibration mode (SNP or INDEL)
+    :param str vcf_id: VCF FileStoreID
+    :param Namespace config: Input parameters and shared FileStoreIDs
+    :return: recalibration table, tranches, plot FileStoreIDs
+    :rtype: tuple
+    """
+    mode = mode.upper()
+    if mode not in ['INDEL', 'SNP']:
```

Comment:
You have two different methods for handling this, why not follow the above method and just have an `else` after you check for `SNP` and `INDEL`?